src/components/Firestore/DocumentPreview/api.tsx

```diff
@@ -42,6 +42,19 @@ export function updateField(
   documentRef.update(...payload);
 }
 
+export async function addFieldToMissingDocument(
+  reference: firestore.DocumentReference,
+  key: string,
+  value: FirestoreAny
+) {
+  const snapshot = await reference.get();
+  if (snapshot.exists) {
+    throw new Error('.set() on an existing document would delete other fields');
```

Comment:
```suggestion
    // TODO: Better surface this to the user.
    throw new Error('.set() on an existing document would delete other fields');
```

Reply:
We should definitely talk about the UX at some point, but until then... this TODO comment will do.